torch_geometric/datasets/metrla.py

```diff
@@ -0,0 +1,177 @@
+        containing the readings, or in download, but download gets skipped
+        after the first time, so that's not reliable. Open to any
+        workarounds, but the dataset has a fixed size, so
+        this is not a deal breaker imo."""
+        self.n_readings = 34272
+
+        self.io = MetrLaIo(n_readings=self.n_readings,
+                           n_previous_steps=n_previous_steps,
+                           n_future_steps=n_future_steps,
+                           add_time_of_day=add_time_of_day,
+                           add_day_of_week=add_day_of_week,
+                           normalized_k=normalized_k)
+        super().__init__(root, transform, pre_transform)
+
+        self.normalized_k = normalized_k
+
+    @property
+    def raw_file_names(self) -> str:
+        r"""The name of the files in the :obj:`self.raw_dir`
+        folder that must be present in order to skip downloading."""
+        return list(self.gdrive_ids.keys())
+
+    @property
+    def processed_file_names(self) -> str:
+        r"""The name of the files in the :obj:`self.processed_dir`
+        folder that must be present in order to skip processing."""
+        return [f'data_{i}.pt' for i in range(self.io.dataset_len)
+                ] + ["edge_index.pt", "edge_attributes.pt"]
+
+    @property
+    def edge_index_file_name(self) -> str:
+        r"""The name of the file containing the edge index."""
+        return self.processed_file_names[-2]
+
+    @property
+    def edge_attributes_file_name(self) -> str:
+        r"""The name of the file containing the edge attributes."""
+        return self.processed_file_names[-1]
+
+    def download(self) -> None:
+        r"""Downloads the dataset to the :obj:`self.raw_dir` folder."""
+        for file_name, gdrive_id in self.gdrive_ids.items():
+            gdd.download_file_from_google_drive(
+                file_id=gdrive_id, dest_path=Path(self.raw_dir) / file_name)
+
+    def process(self) -> None:
+        r"""Processes the dataset to the :obj:`self.processed_dir` folder."""
+
+        x, y = self.io.get_metrla_data(data_path=self.raw_paths[3])
+
+        adjacency_matrix = self.io.generate_adjacency_matrix(
+            distances_path=self.raw_paths[0],
+            sensor_ids_path=self.raw_paths[2])
+
+        adjacency_matrix = torch.from_numpy(adjacency_matrix)
+
+        edge_index, edge_attributes = dense_to_sparse(adjacency_matrix)
+
+        edge_index_path = Path(self.processed_dir) / self.edge_index_file_name
+        edge_attributes_path = Path(
+            self.processed_dir) / self.edge_attributes_file_name
+
+        torch.save(obj=edge_index, f=edge_index_path)
+        torch.save(obj=edge_attributes, f=edge_attributes_path)
+
+        for idx in range(self.io.dataset_len):
+            # Select the "slice" among the first dimension
+            # x_index (n_previous_steps, n_nodes, n_features)
+            x_index = torch.tensor(data=x[idx, ...], dtype=torch.float32)
```

Comment:
```suggestion
            x_index = x[idx]
```
Isn't that sufficient?

Reply:
That's just to convert from numpy to torch tensor, since the Data constructor specifies x to be Optional[Tensor].